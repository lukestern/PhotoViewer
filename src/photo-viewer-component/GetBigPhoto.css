.large-photo-container {
    display: flex;
    width: 100%;
}

.large-photo {
    width: 90vw;
    margin: 5vw;
}