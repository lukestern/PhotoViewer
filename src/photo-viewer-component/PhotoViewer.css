* {
    font-family: 'Segoe UI', 'Open Sans';
}

h1 {
    color: blueviolet;
}

.small-photo-container {
    display: flex;
    justify-content: space-between;
}

.small-photo {
    width: 23vw;
}