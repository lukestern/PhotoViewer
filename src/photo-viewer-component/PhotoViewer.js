import React from "react";  // import React (to provide access to JSX)

function PhotoViewer() {    // declare a new function called 'PhotoViewer'
    return (                
        <div className="small-photo-container">              
            <img className="small-photo" src="https://picsum.photos/seed/picsum/500/300" />
            <img className="small-photo" src="https://picsum.photos/seed/picsum/500/300" />
            <img className="small-photo" src="https://picsum.photos/seed/picsum/500/300" />
            <img className="small-photo" src="https://picsum.photos/seed/picsum/500/300" />
        </div>
    );
}

export {PhotoViewer};       // Export the function so that it can be imported in other files