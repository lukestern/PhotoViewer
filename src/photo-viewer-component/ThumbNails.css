.small-photo-container {
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
}

.small-photo {
    width: 22vw;
    height: 12vw;
    margin: 1vw;
}

.small-photo-selected {
    border: 0.25vw solid blue;
    width: 21.75vw;
    height: 11.75vw;
    margin: 1vw;
}