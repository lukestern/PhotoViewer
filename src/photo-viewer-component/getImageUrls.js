export const brokenImages = [
    1, 24, 32, 36, 44, 47
];

export function getImageUrls() {
    const photoIdList = [];

    for (let i = 0; i < 50; i++) {
        if (!brokenImages.includes(i)) {
            const imageNumberString = i.toString().padStart(2, '0');
            photoIdList.push(imageNumberString);
        }
    }

    return photoIdList;
}