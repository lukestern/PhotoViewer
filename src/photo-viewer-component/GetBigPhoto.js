import React from "react";
import './GetBigPhoto.css';

export function GetBigPhoto(props) {
    return (
        <div className="large-photo-container">
            <img alt='' className="large-photo" src={props.photoSelectedUrl}  onClick={() => props.setPhotoSelected(null)} />
        </div>
    );
}