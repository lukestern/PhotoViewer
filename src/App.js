import { ThumbNails } from './photo-viewer-component/ThumbNails';
import { GetBigPhoto } from './photo-viewer-component/GetBigPhoto';
import './App.css';
import { useState } from "react";


function App() {
  const [photoSelectedUrl, setPhotoSelected] = useState(null);

  return (
    <div>
      <h1>React Photo Viewer</h1>
      <GetBigPhoto photoSelectedUrl={photoSelectedUrl} />
      <ThumbNails photoSelectedUrl={photoSelectedUrl} setPhotoSelected={setPhotoSelected} />
    </div>
  );
}

export default App;
