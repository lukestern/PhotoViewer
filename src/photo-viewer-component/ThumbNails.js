import React from "react";
import './ThumbNails.css';
import { getImageUrls } from './getImageUrls'



export function ThumbNails (props) {
    const urlList = getImageUrls();
    const imgList = urlList.map(
        (url) =>
            <img 
                key={url}
                alt=''
                className={url === props.photoSelectedUrl ? "small-photo-selected" : "small-photo"}
                onClick={() => url === props.photoSelectedUrl ? props.setPhotoSelected(null) : props.setPhotoSelected(url)}
                src={url} />
    );

    return (
        <div className="small-photo-container">
            {imgList}
        </div>
    );
}