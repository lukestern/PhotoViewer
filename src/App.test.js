import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import { getImageUrls, brokenImages } from './photo-viewer-component/getImageUrls';
import { GetBigPhoto } from './photo-viewer-component/GetBigPhoto';
import App from './App';
import { ThumbNails } from './photo-viewer-component/ThumbNails';
import renderer from 'react-test-renderer';

// Unit tests
test('renders react photo viewer heading', () => {
  render(<App />);
  const linkElement = screen.getByText(/React Photo Viewer/i);
  expect(linkElement).toBeInTheDocument();
});

test('image urls are correctly generated', () => {
  const imageUrls = getImageUrls();
  const brokenUrls = brokenImages.map(
    (imageNumber) => `https://picsum.photos/id/6${imageNumber}/1600/900.jpg`
  );
  imageUrls.forEach(
    (url) => expect(!brokenUrls.includes(url))
  );
});

test('image is selected when it is clicked', () => {
  render(<App />);

  const testedUrl = getImageUrls()[0];
  const testedImage = screen.getByAltText(testedUrl);
  userEvent.click(testedImage);
  expect(testedImage.className === 'small-photo-selected');

  const testedUrl1 = getImageUrls()[10];
  const testedImage1 = screen.getByAltText(testedUrl1);
  userEvent.click(testedImage1);
  expect(testedImage1.className === 'small-photo-selected');
});

test('large image is displayed if a thumbnail is selected', () => {
  render(<App />);

  const testedUrl = getImageUrls()[0];
  const testedImage = screen.getByAltText(testedUrl);

  userEvent.click(testedImage);

  expect(testedImage.className === 'large-photo');
});

test('regression test for big photo', () => {
  const photoUrl = "Hello"
  render(<GetBigPhoto photoSelectedUrl={photoUrl} setPhotoSelected={() => { }} />);

  const bigPhoto = screen.getByTestId('large-photo');

  expect(bigPhoto).toBeInTheDocument();
  expect(bigPhoto.src === photoUrl);
  expect(bigPhoto.className === 'large-photo');
});

test('regression test for thumbnails', () => {
  const photoUrl = "Hello"
  render(<ThumbNails photoSelectedUrl={photoUrl} setPhotoSelected={() => { }} />)

  const smallPhotoContainer = screen.getByTestId('small-photo-container');

  expect(smallPhotoContainer).toBeInTheDocument();
  expect(smallPhotoContainer.children).toHaveLength(44);
});

// Snapshot tests
describe("PhotoViewer", () => {
  let tree;

  beforeEach(() => {
    tree = renderer.create(<App />)
  })

  it("should match snapshots without selected photo", async () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("GetBigPhoto should match snapshots without selected photo", async () => {
    renderer.act(() => {
      tree.update(<GetBigPhoto photoSelectedUrl={null} setPhotoSelected={null} />)
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("GetBigPhoto should match snapshots with selected photo", async () => {
    renderer.act(() => {
      tree.update(<GetBigPhoto photoSelectedUrl={"Test"} setPhotoSelected={() => { }} />)
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("ThumbNails should match snapshots without selected photo", async () => {
    renderer.act(() => {
      tree.update(<ThumbNails photoSelectedUrl={null} setPhotoSelected={null} />)
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("ThumbNails should match snapshots with selected photo", async () => {
    renderer.act(() => {
      tree.update(<ThumbNails photoSelectedUrl={"Test"} setPhotoSelected={() => { }} />)
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});