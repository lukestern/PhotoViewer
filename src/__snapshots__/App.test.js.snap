// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoViewer GetBigPhoto should match snapshots with selected photo 1`] = `
<div
  className="large-photo-container"
  data-testid="large-photo"
>
  <img
    alt="Test"
    className="large-photo"
    onClick={[Function]}
    src="Test"
  />
</div>
`;

exports[`PhotoViewer GetBigPhoto should match snapshots without selected photo 1`] = `
<div
  className="large-photo-container"
  data-testid="large-photo"
>
  <img
    alt={null}
    className="large-photo"
    onClick={[Function]}
    src={null}
  />
</div>
`;

exports[`PhotoViewer ThumbNails should match snapshots with selected photo 1`] = `
<div
  className="small-photo-container"
  data-testid="small-photo-container"
>
  <img
    alt="https://picsum.photos/id/600/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/600/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/602/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/602/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/603/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/603/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/604/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/604/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/605/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/605/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/606/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/606/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/607/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/607/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/608/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/608/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/609/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/609/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/610/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/610/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/611/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/611/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/612/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/612/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/613/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/613/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/614/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/614/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/615/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/615/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/616/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/616/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/617/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/617/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/618/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/618/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/619/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/619/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/620/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/620/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/621/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/621/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/622/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/622/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/623/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/623/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/625/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/625/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/626/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/626/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/627/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/627/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/628/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/628/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/629/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/629/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/630/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/630/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/631/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/631/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/633/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/633/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/634/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/634/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/635/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/635/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/637/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/637/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/638/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/638/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/639/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/639/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/640/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/640/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/641/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/641/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/642/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/642/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/643/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/643/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/645/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/645/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/646/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/646/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/648/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/648/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/649/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/649/1600/900.jpg"
  />
</div>
`;

exports[`PhotoViewer ThumbNails should match snapshots without selected photo 1`] = `
<div
  className="small-photo-container"
  data-testid="small-photo-container"
>
  <img
    alt="https://picsum.photos/id/600/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/600/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/602/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/602/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/603/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/603/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/604/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/604/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/605/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/605/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/606/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/606/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/607/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/607/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/608/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/608/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/609/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/609/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/610/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/610/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/611/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/611/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/612/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/612/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/613/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/613/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/614/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/614/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/615/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/615/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/616/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/616/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/617/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/617/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/618/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/618/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/619/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/619/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/620/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/620/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/621/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/621/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/622/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/622/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/623/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/623/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/625/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/625/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/626/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/626/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/627/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/627/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/628/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/628/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/629/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/629/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/630/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/630/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/631/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/631/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/633/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/633/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/634/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/634/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/635/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/635/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/637/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/637/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/638/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/638/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/639/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/639/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/640/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/640/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/641/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/641/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/642/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/642/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/643/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/643/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/645/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/645/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/646/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/646/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/648/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/648/1600/900.jpg"
  />
  <img
    alt="https://picsum.photos/id/649/1600/900.jpg"
    className="small-photo"
    onClick={[Function]}
    src="https://picsum.photos/id/649/1600/900.jpg"
  />
</div>
`;

exports[`PhotoViewer should match snapshots without selected photo 1`] = `
<div>
  <h1>
    React Photo Viewer
  </h1>
  <div
    className="large-photo-container"
    data-testid="large-photo"
  >
    <img
      alt={null}
      className="large-photo"
      onClick={[Function]}
      src={null}
    />
  </div>
  <div
    className="small-photo-container"
    data-testid="small-photo-container"
  >
    <img
      alt="https://picsum.photos/id/600/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/600/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/602/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/602/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/603/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/603/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/604/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/604/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/605/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/605/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/606/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/606/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/607/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/607/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/608/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/608/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/609/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/609/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/610/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/610/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/611/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/611/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/612/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/612/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/613/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/613/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/614/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/614/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/615/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/615/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/616/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/616/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/617/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/617/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/618/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/618/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/619/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/619/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/620/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/620/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/621/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/621/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/622/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/622/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/623/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/623/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/625/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/625/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/626/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/626/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/627/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/627/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/628/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/628/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/629/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/629/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/630/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/630/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/631/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/631/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/633/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/633/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/634/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/634/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/635/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/635/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/637/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/637/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/638/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/638/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/639/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/639/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/640/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/640/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/641/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/641/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/642/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/642/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/643/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/643/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/645/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/645/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/646/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/646/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/648/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/648/1600/900.jpg"
    />
    <img
      alt="https://picsum.photos/id/649/1600/900.jpg"
      className="small-photo"
      onClick={[Function]}
      src="https://picsum.photos/id/649/1600/900.jpg"
    />
  </div>
</div>
`;
